Userland/Libraries/LibWeb/DOM/ShadowRoot.cpp

```diff
@@ -18,9 +19,15 @@ ShadowRoot::ShadowRoot(Document& document, Element& host, Bindings::ShadowRootMo
     : DocumentFragment(document)
     , m_mode(mode)
 {
+    document.register_shadow_root({}, *this);
     set_host(&host);
 }
 
+ShadowRoot::~ShadowRoot()
+{
+    document().unregister_shadow_root({}, *this);
```

Comment:
This needs to be in `finalize()` since `document()` may have been destroyed by the time we run `~ShadowRoot()`

Reply:
Fixed